src/components/FilterPanel/FilterPanelCheckboxWithOverflowMenu/FilterPanelCheckboxWithOverflowMenu.js

```diff
@@ -0,0 +1,152 @@
+  const [showOverflowButton, setShowOverflowButton] = React.useState(false);
+  const [overflowIsOpen, setOverflowIsOpen] = React.useState(false);
+  const { createFocusHandler, createBlurHandler } = useComponentFocus(10);
+
+  const handleFocus = createFocusHandler(() => {
+    setShowOverflowButton(true);
+  });
+
+  const handleBlur = createBlurHandler(() => {
+    setShowOverflowButton(false);
+  });
+
+  /**
+   * Sets the width of the overflow menu to match the width of this component's width and adjusts
+   * its position so it is right alingned with the overflow menu button.
+   *
+   * @param {HTMLElement} menuBody Element containing the overflow menu options.
+   * @param {string} direction The menu position relative to the menu button. Not used.
+   * @param {HTMLElement} menuButton Overflow menu button element.
+   * @returns {{top: number, left: number}} The menu offset measurements.
+   */
+  const updateMenuWidthAndSetOffset = (menuBody, direction, menuButton) => {
+    const width = containerRef.current.clientWidth;
+    const left = width / 2 - width + menuButton.clientWidth / 2;
+
+    // Set the menu's width.
+    // eslint-disable-next-line no-param-reassign
+    menuBody.style.width = `${width}px`;
+
+    // Set the menu's left position to match the return left position value. We need to do this so
+    // that the overflow menu is initially positioned properly when the menu is too close to the
+    // edge of the viewport.
+    if (!overflowIsOpen) {
+      // eslint-disable-next-line no-param-reassign
+      menuBody.style.left = `${left}px`;
+    }
+
+    return {
+      top: 0,
+      left,
+    };
+  };
+
+  return (
+    <div
+      className={classnames(className, namespace, {
+        [`${namespace}--open`]: overflowIsOpen,
+      })}
+      onMouseOver={() => setShowOverflowButton(true)}
+      onMouseLeave={() => !overflowIsOpen && setShowOverflowButton(false)}
+      onFocus={handleFocus}
+      onBlur={handleBlur}
+      ref={containerRef}
+    >
+      <FilterPanelCheckbox
+        className={checkboxClassName}
+        wrapperClassName={classnames(
+          checkboxWrapperClassName,
+          `${namespace}__wrapper`
+        )}
+        {...other}
+      />
+      {showOverflowButton && (
+        <OverflowMenu
+          className={classnames(
+            overflowMenuClassName,
+            `${namespace}__overflow-button`
+          )}
+          menuOptionsClass={classnames(
+            overflowMenuOptionsClassName,
+            `${namespace}__overflow-options`
+          )}
+          menuOffsetFlip={updateMenuWidthAndSetOffset}
+          ariaLabel={overflowMenuAriaLabel}
+          onOpen={() => setOverflowIsOpen(true)}
+          onClose={() => setOverflowIsOpen(false)}
+          flipped
+        >
+          {children}
+        </OverflowMenu>
+      )}
+    </div>
+  );
+};
+
+FilterPanelCheckboxWithOverflowMenu.propTypes = {
+  ...FilterPanelCheckbox.propTypes,
```

Comment:
👏 I like this because we can maintain parity with the `FilterPanelCheckbox` in case that API changes!

The only thing to note is that Storybook's documentation generation doesn't like this so much. 😭 We could potentially fill in the gaps by linking out to the `FilterPanelCheckbox`stories from the `FilterPanelCheckboxWithOverflowMenu`, similarly to https://ibm-security.carbondesignsystem.com/?path=/story/components-accordion--default 's 'Show Info'.

Reply:
Ah yes, I can do that.